### PR TITLE
Fix for issue 7451: Saving library doesn’t create .bib file but a file without ending

### LINKED ISSUE #7451: Saving library doesn't create .bib file but a file without ending
File > New library and then Save library > choose name > lower right corner choose BIBTEX_DB file (only option) > Save
or File > Save library as > choose name > lower right corner choose BIBTEX_DB file (only option) > Save
File is saved without any ending. Expected to be saved as .bib instead, as none(?) other file-endings are displayed when using File > Open library.

Workaround: add .bib manually when choosing name.

JabRef 5.2--2020-12-24--6a2a512
Linux 5.8.0-43-generic amd64 
Java 15.0.1

### PR DESCRIPTION
**Pull request by guyf2010**
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

Workaround for issue #7451 . Solution implemented with ryanhore23 based on Siedlerchr's suggested solution.

Fixes #7451 

Changes:

After file save dialog is completed, the filename is checked. If it already ends in ".bib" (Case insensitive) it continues as previously. Otherwise it appends ".bib" to the end of the selected path. It will return null if this file already exists to prevent any chance of data loss. In the event it cancels, it displays a notification through the dialogService that saving failed.

### CODE REVIEW

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java [hunk 1]
```diff
@@ -161,6 +162,19 @@ boolean saveAs(Path file, SaveDatabaseMode mode) {
...
+        if(selectedPath.isPresent()){
+            Path savePath = selectedPath.get();
+            // Workaround for linux systems not adding file extension
+            if(!(savePath.getFileName().toString().toLowerCase().endsWith(".bib"))){
+                savePath = Path.of(savePath.toString()+".bib");
+                if(Files.notExists(savePath)){
+                  selectedPath = Optional.of(savePath);
+                } else {
+                  selectedPath = Optional.empty();
+                  dialogService.notify(Localization.lang("Could not save file."));
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java:174]: As a User I should have the option to confirm that I want to overwrite the file if it exists

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
Codewise looks good so far, I think you should add an option to allow overwriting the file

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java (hunk 1 above)
Comment (guyf2010) [reply src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java:174]: That seems like a pretty reasonable request. Do you have any recommendations on where to look for UI popups keeping in theme with JabRef? I haven't used much of Java's UI features previously. Any documentation or advice is much appreciated.

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java:174]: Just look in Jabref's dialogService methods: dialogService.showConfirmationDialogAndWait(...)
You will find an example somewhere in the code

File: CHANGELOG.md [hunk 2]
```diff
@@ -90,6 +91,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where the article title with colon fails to download the arXiv link (pdf file). [#7660](https://github.com/JabRef/issues/7660)
 - We fixed an issue where the keybinding for delete entry did not work on the main table [7580](https://github.com/JabRef/jabref/pull/7580)
 - We fixed an issue where the RFC fetcher is not compatible with the draft [7305](https://github.com/JabRef/jabref/issues/7305)
+ main
```
Comment (Siedlerchr) [inline CHANGELOG.md:94]:
```suggestion

```

**Review by Siedlerchr — APPROVED** [review]

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java [hunk 3]
```diff
@@ -161,6 +162,19 @@ boolean saveAs(Path file, SaveDatabaseMode mode) {
                 .build();
         Optional<Path> selectedPath = dialogService.showFileSaveDialog(fileDialogConfiguration);
         selectedPath.ifPresent(path -> preferences.setWorkingDirectory(path.getParent()));
+        if (selectedPath.isPresent()) {
+            Path savePath = selectedPath.get();
+            // Workaround for linux systems not adding file extension
+            if (!(savePath.getFileName().toString().toLowerCase().endsWith(".bib"))) {
```
Comment (koppor) [inline src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java:168]: I was beginning to use Java optionals. However, it is longer than expected.

Thus, it is OK for me to have it like it is and NOT do as follows

(not having same semantics)

```java
            if (getFileExtension(file).map(String::toLowerCase).filter(extension -> extension.equals("bib")).ifpresent()) {
```

**Review by koppor — APPROVED** [review]

**Pull request merged**